doc/release/upcoming_changes/27883.change.rst

```diff
@@ -0,0 +1,18 @@
+Changes to the main iterator and potential numerical changes
+------------------------------------------------------------
+The main iterator used in math functions, for ``np.nditer`` and
+on the ``NpyIter`` in the C-API now behaves different for some
```

Comment:
Maybe "via ``np.nditer`` in Python and ``NpyIter`` in C"

Also different -> differently